src/super_gradients/training/utils/media/video.py

```diff
@@ -30,6 +30,15 @@ def load_video(file_path: str, max_frames: Optional[int] = None) -> Tuple[List[n
     return frames, fps
 
 
+def lazy_load_video(file_path: str, max_frames: Optional[int] = None) -> Tuple[Generator, int]:
```

Comment:
Function type annotation does not match real return value (Tuple of 2 vs 3)

Reply:
Also I would replace `Generator` with one of the two
- `Generator[np.ndarray, None, None]` most standard typing for generators, but too verbose IMO
- `Iterator[np.ndarray]` I prefer this. It is mentioned in the python doc so I think it's fair to use https://docs.python.org/3.6/library/typing.html#typing.Generator